### PR TITLE
Let `ApiKeyCommand` use and prefer a long lived api key `HEXPM_API_KEY`

### PR DESCRIPTION
**Pull request by Skenvy**
Addresses #2844 

Add a `HEXPM_API_KEY` which `ApiKeyCommand` will prefer, but will check if it's supplied or not and fall back to requesting the username and password, unless either the api key or username AND password are supplied via the env vars, in which case, it will prefer the api key. But if an API key isn't supplied to the prompt and a username and password are supplied, it will use those, and only prompt for them if no API key is given to the prompt for one if no credentials are supplied from env vars. Then wrap the creating and removing of the temporary key produced using the username and password with the check from whether or not a long lived api key was provided.

### CODE REVIEW

**Review by lpil — COMMENTED** [review]
Hello! Thank you for this.

I'm finding it a bit hard to follow this code as there's a lot of boolean logic in one function, and I struggle a bit with boolean blindless.

Could we do something like this instead:

```rust
let key = var(API_KEY_KEY).unwrap_or_default().trim();
if key.is_empty() {
  self.with_new_api_key(...)
} else {
  self.with_api_key(...)
}
```

Then the two paths have been split into two functions, making it easier to follow each one.

**Comment by Skenvy** [comment]
The only problem then is that without just moving the same logic inside `with_new_api_key`, we wouldn't be able to check if the username and password are supplied or not before checking if it's a good idea to prompt for the api key before prompting for them or not, i.e. the api key will never be prompted for, if you're happy with that.

**Comment by lpil** [comment]
Yup let's use environment variables for the key always.

**Comment by Skenvy** [comment]
I'm not particularly rusty, is there a preference between;
```rust
if api_key.is_empty() {
    let result = self.with_new_api_key(runtime.handle(), &hex_config);
} else {
    let result = self.with_api_key(runtime.handle(), &hex_config, &api_key);
}

result
```
and
```rust
if api_key.is_empty() {
    self.with_new_api_key(runtime.handle(), &hex_config)
} else {
    self.with_api_key(runtime.handle(), &hex_config, &api_key)
}
```

**Comment by Skenvy** [comment]
Ok it compiles and works as expected now (publishes / retires with the key and user+pass), and only prompts for user/pass with no api key given.

**Comment by lpil** [comment]
> I'm not particularly rusty, is there a preference between;
> 
> ```rust
> if api_key.is_empty() {
>     let result = self.with_new_api_key(runtime.handle(), &hex_config);
> } else {
>     let result = self.with_api_key(runtime.handle(), &hex_config, &api_key);
> }
> 
> result
> ```
> 
> and
> 
> ```rust
> if api_key.is_empty() {
>     self.with_new_api_key(runtime.handle(), &hex_config)
> } else {
>     self.with_api_key(runtime.handle(), &hex_config, &api_key)
> }
> ```

The first is a compile error as the variable `result` isn't in scope any more after the block. The other looks good 👍

**Review by lpil — APPROVED** [review]
Great! Thank you.

Would you mind updating the changelog? Thank you
